xarray/tests/test_cftime_offsets.py

```diff
@@ -1541,3 +1602,26 @@ def test_to_offset_deprecation_warning(freq):
     # Test for deprecations outlined in GitHub issue #8394
     with pytest.warns(FutureWarning, match="is deprecated"):
         to_offset(freq)
+
+
+@pytest.mark.filterwarnings("ignore:Converting a CFTimeIndex with:")
+@pytest.mark.parametrize("start", ("2000", "2001"))
+@pytest.mark.parametrize("end", ("2000", "2001"))
+@pytest.mark.parametrize("freq", ("MS", "-1MS", "YS", "-1YS"))
+def test_cftime_range_same_as_pandas(start, end, freq):
+    result = date_range(start, end, freq=freq, calendar="standard", use_cftime=True)
+    result = result.to_datetimeindex()
+    expected = date_range(start, end, freq=freq, use_cftime=False)
+
+    np.testing.assert_array_equal(result, expected)
+
+
+@pytest.mark.parametrize("start", ("2000", "2001"))
+@pytest.mark.parametrize("end", ("2000", "2001"))
+@pytest.mark.parametrize("freq", ("M", "-1M", "Y", "-1Y"))
+def test_cftime_range_end_same_len_pandas(start, end, freq):
+    result = date_range(start, end, freq=freq, calendar="standard", use_cftime=True)
+    expected = date_range(start, end, freq=freq, use_cftime=False)
+
+    # end of month/ year is not convertable to datetimeindex
+    assert result.size == expected.size
```

Comment:
I think we can check for equality now, similar to the test above:
```suggestion
    result = date_range(start, end, freq=freq, calendar="standard", use_cftime=True)
    result = result.to_datetimeindex()
    expected = date_range(start, end, freq=freq, use_cftime=False)

    np.testing.assert_array_equal(result, expected)
```

Reply:
Ultimately I suppose this test could now be merged with the one above—i.e. just merge the frequency strings.

Reply:
of course - thanks